### PR TITLE
Tools/web-development_Kojima_homework

### PR DESCRIPTION
**Pull request by kojiyuu-123**
Hello, please check my homework at "Tools" of "web-development".
To run all programs, please run "bash_file_5.sh"

The details of these programs are blow.
"bash_file.sh" : Display some words.
"bash_file_2.sh" : Execute some commands.
"bash_file_3.sh" : Search the word, "character".
"bash_file_4.sh" : Count the number of the word, "character".
"bash_file_5.sh" : Clone above files , use "git pull" to fetch each of them and run above programs.

Thank you.

### CODE REVIEW

File: Tools_webdevelopment_Homework/bash_file_5.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+chmod 777 "bash_file_5.sh"
+
+array=(bash_file/bash_file.sh bash_file_2/bash_file_2.sh bash_file_3/bash_file_3.sh bash_file_4/bash_file_4.sh)
+
+for i in ${array[@]}
+do
+	./$i
+	echo -e ""
```
Comment (ohralathe) [inline Tools_webdevelopment_Homework/bash_file_5.sh:8]: just print? the requirement is to fetch for each? right?

File: Tools_webdevelopment_Homework/bash_file_5.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+chmod 777 "bash_file_5.sh"
+
+array=(bash_file/bash_file.sh bash_file_2/bash_file_2.sh bash_file_3/bash_file_3.sh bash_file_4/bash_file_4.sh)
```
Comment (ohralathe) [inline Tools_webdevelopment_Homework/bash_file_5.sh:3]: why don't you put the repo URL?

File: Tools_webdevelopment_Homeworks/bash_file.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#Set a bash file's execute permission
```
Comment (ohralathe) [inline Tools_webdevelopment_Homeworks/bash_file.sh:1]: from this file to the end. Does it duplicate?

File: Tools_webdevelopment_Homework/bash_file/bash_file.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#Set a bash file's execute permission
+chmod 777 "bash_file/bash_file.sh"
```
[the remark this answers is not in the document]
Comment (kojiyuu-123) [reply Tools_webdevelopment_Homework/bash_file/bash_file.sh:2]: Yes, all users can read ,write and execute this file.

File: Tools_webdevelopment_Homeworks/bash_file.sh (hunk 3 above)
Comment (kojiyuu-123) [reply Tools_webdevelopment_Homeworks/bash_file.sh:1]: Yes. I pushed these files before modifying

File: Tools_webdevelopment_Homework/bash_file_5.sh (hunk 1 above)
Comment (kojiyuu-123) [reply Tools_webdevelopment_Homework/bash_file_5.sh:8]: I overlooked the requirement. I should use "git fetch", right?

File: Tools_webdevelopment_Homework/bash_file_5.sh (hunk 2 above)
Comment (kojiyuu-123) [reply Tools_webdevelopment_Homework/bash_file_5.sh:3]: I misunderstood the requirement, "put all your repos to an array".
I'll modify this program.

File: Tools_webdevelopment_Homework/bash_file/bash_file.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (ohralathe) [reply Tools_webdevelopment_Homework/bash_file/bash_file.sh:2]: @kojiyuu-123 this is so dangerous, it like you said, hey this is a `lock` and here is `my key`, use it if you want.

you got my idea?

File: Tools_webdevelopment_Homework/bash_file_5.sh (hunk 1 above)
Comment (ohralathe) [reply Tools_webdevelopment_Homework/bash_file_5.sh:20]: yes.

File: Tools_webdevelopment_Homework/bash_file_5.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
+chmod 777 "bash_file_5.sh"
+
+# clone repositories
+url_array=(https://github.com/kojiyuu-123/bash_file.git https://github.com/kojiyuu-123/bash_file_2.git https://github.com/kojiyuu-123/bash_file_3.git https://github.com/kojiyuu-123/bash_file_4.git)
```
Comment (ohralathe) [inline Tools_webdevelopment_Homework/bash_file_5.sh:4]: is there any way better?
see: `https://github.com/kojiyuu-123/` repeated all the cases

File: Tools_webdevelopment_Homework/bash_file_5.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+# fetch all repositories
+array=(bash_file bash_file_2 bash_file_3 bash_file_4)
+for j in ${array[@]};
+do
+	cd ${j}
+	git checkout master
+	git pull origin master
+	cd ../
+	./${j}/bash_file*.sh
```
Comment (ohralathe) [inline Tools_webdevelopment_Homework/bash_file_5.sh:19]: what does it mean?

File: Tools_webdevelopment_Homework/bash_file_5.sh (hunk 1 above)
Comment (ohralathe) [reply Tools_webdevelopment_Homework/bash_file_5.sh:20]: yes

File: Tools_webdevelopment_Homeworks/bash_file.sh (hunk 3 above)
Comment (ohralathe) [reply Tools_webdevelopment_Homeworks/bash_file.sh:1]: why do you not remove it?

File: Tools_webdevelopment_Homework/bash_file/bash_file.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (kojiyuu-123) [reply Tools_webdevelopment_Homework/bash_file/bash_file.sh:2]: It's so dangerous. I'll change "777" to "755".

File: Tools_webdevelopment_Homeworks/bash_file.sh (hunk 3 above)
Comment (kojiyuu-123) [reply Tools_webdevelopment_Homeworks/bash_file.sh:1]: I don't know the system of GitHub. So, I'll look up the way to remove it and do.

File: Tools_webdevelopment_Homework/bash_file_5.sh (hunk 5 above)
Comment (kojiyuu-123) [reply Tools_webdevelopment_Homework/bash_file_5.sh:4]: I should assign the sentence , https://github.com/kojiyuu-123, to variable.

File: Tools_webdevelopment_Homework/bash_file_5.sh (hunk 6 above)
Comment (kojiyuu-123) [reply Tools_webdevelopment_Homework/bash_file_5.sh:19]: Each repos have "master" branch. "master" branch is main branch. So checkout the branch to "master", and use "git pull" to fetch  the repos.
And it runs "bash_file_xxx.sh".

Comment (ohralathe) [reply Tools_webdevelopment_Homework/bash_file_5.sh:20]: what is `bash_file*.sh`? where is the repo URL that I used in the exercises, this homework require you refactor that exercise, right?

Could you please write a *.sh file which I can run that file and do all the jobs (fetch all repo), `NOT the PSEUDOCODE`

Comment (kojiyuu-123) [reply Tools_webdevelopment_Homework/bash_file_5.sh]: As I fixed "bash_file_5.sh", you can run this.
I don't understand the meaning "refactor" in this requirement.

File: Tools_webdevelopment_Homework/bash_file_5.sh [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+repos_str="https://github.com/kojiyuu-123"
+url_array=(${repos_str}/bash_file.git ${repos_str}/bash_file_2.git ${repos_str}/bash_file_3.git ${repos_str}/bash_file_4.git)
+
+for i in ${url_array[@]};
+do
+	git clone ${i}
+done
+
+# fetch all repositories
+array=(bash_file bash_file_2 bash_file_3 bash_file_4)
```
Comment (ohralathe) [inline Tools_webdevelopment_Homework/bash_file_5.sh:13]: @kojiyuu-123 there is no update anymore?
can you run this file?

Comment (kojiyuu-123) [reply Tools_webdevelopment_Homework/bash_file_5.sh:13]: yes.
I was able to run this file.

**Pull request closed without merging**